tests/rptest/tests/tiered_storage_model_test.py

```diff
@@ -0,0 +1,551 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+import dataclasses
+import json
+import random
+import re
+import time
+from concurrent.futures import ThreadPoolExecutor, Future
+from threading import Condition
+from collections import defaultdict
+
+from ducktape.mark import matrix
+from ducktape.tests.test import TestContext
+from ducktape.utils.util import wait_until
+
+from rptest.clients.kafka_cli_tools import KafkaCliTools
+from rptest.clients.kafka_cat import KafkaCat
+from rptest.clients.rpk import RpkTool
+from rptest.clients.types import TopicSpec
+from rptest.services.action_injector import random_process_kills
+from rptest.services.redpanda import RedpandaService
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.admin import Admin
+from rptest.services.cluster import cluster
+from rptest.services.kgo_verifier_services import KgoVerifierConsumerGroupConsumer, KgoVerifierProducer, KgoVerifierRandomConsumer, KgoVerifierSeqConsumer
+from rptest.services.redpanda import SISettings, get_cloud_storage_type, make_redpanda_service, CHAOS_LOG_ALLOW_LIST, MetricsEndpoint
+from rptest.utils.si_utils import nodes_report_cloud_segments, BucketView, NTP
+from rptest.tests.tiered_storage_model import TestCase, TieredStorageEndToEndTest, get_tiered_storage_test_cases, TestRunStage, CONFIDENCE_THRESHOLD
+
+MAX_RETRIES = 5
+
+
+class TieredStorageTest(TieredStorageEndToEndTest, RedpandaTest):
+
+    # Topic doesn't have tiered storage enabled by default
+    topics = (
+        TopicSpec(
+            partition_count=1,  # TODO: maybe use more partitions
```

Comment:
The comment raises a good point. Would we have a separate model for each partition?

Reply:
no, we will be able to bump number of partitions, but for now it's easier to go with one partition